### PR TITLE
Revert back to OCaml 5.1.1

### PR DESCRIPTION
**Pull request by feihong**
- [x] Go back to OCaml 5.1.1 from OCaml 5.2.0 because [lack of type-on-hover](https://github.com/reasonml/reason-react/issues/840) is really annoying
- [x] Update deprecated OCaml Manual URLs to their modern equivalents
- [x] Change "polymorphic variant constructor" to "variant tag" to match the terminology used in [OCaml docs](https://ocaml.org/manual/5.2/polyvariant.html)

### CODE REVIEW

File: docs/styling-with-css/index.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ line to the top of `Order.re`:
 ```
 
 The `{||}` string literal is known as a [quoted
-string literal](https://v2.ocaml.org/manual/lex.html#sss:stringliterals), and it
+string literal](https://ocaml.org/manual/5.2/lex.html#sss:stringliterals), and it
```
Comment (jchavarri) [inline docs/styling-with-css/index.md:29]: I was a bit confused by this kind of change, because on one hand the PR is reverting to 5.1.1 but on the other it's picking 5.2 urls.

I understand both changes are unrelated?

**Review by jchavarri — APPROVED** [review]

**Comment by jchavarri** [comment]
>  Update deprecated OCaml Manual URLs to their modern equivalents

Can use Ahrefs' Site Audit to find out broken links ✨

File: docs/styling-with-css/index.md (hunk 1 above)
Comment (feihong) [reply docs/styling-with-css/index.md:29]: Yeah, I don't think linking to 5.2 URLs here causes any problems because there weren't any changes to the manual in those parts.
